### ISSUE TITLE
test(chainrules): add HVP test with DAG merge point to cover accumulate branches

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/tenferro-rs#170]
## Context

`chainrules/src/lib.rs` is at 95.4% coverage. The uncovered lines (448, 458) are the `Some(existing)` branches in `Tape::hvp`'s cotangent/cotangent-tangent accumulation loop:

```rust
// line 446-452
match cotangents[idx].take() {
    Some(existing) => {
        cotangents[idx] = Some(V::accumulate_tangent(existing, &grad));  // line 448: UNCOVERED
    }
    None => {
        cotangents[idx] = Some(grad);
    }
}

// line 456-462
match cot_tangents[idx].take() {
    Some(existing) => {
        cot_tangents[idx] = Some(V::accumulate_tangent(existing, &grad_tan));  // line 458: UNCOVERED
    }
    None => {
        cot_tangents[idx] = Some(grad_tan);
    }
}
```

These branches are reached only when the computation graph has a **merge point** — a node whose output is consumed by multiple downstream operations, causing multiple cotangent contributions to accumulate at that node during reverse-mode traversal.

## What to do

Add a HVP test where the same tracked tensor is used in two or more operations (e.g., `loss = einsum(A, B) + einsum(A, C)` where `A` appears twice), so that `A`'s node receives cotangent contributions from both paths during `hvp`.

## Expected outcome

Lines 448 and 458 become covered, bringing `chainrules/src/lib.rs` to ~100%.
